src/lib/localization-hoc.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+
+import ConnectedIntlProvider from './connected-intl-provider.jsx';
+
+/*
+ * Higher Order Component to provide localiztion state. Creates a nested IntlProvider
+ * to handle Gui intl context. The component accepts an onSetLanguage callback that is
+ * called when the locale chagnes.
+ * @param {React.Component} WrappedComponent - component to provide state for
+ * @returns {React.Component} component with intl state provided from redux
+ */
+const LocalizationHOC = function (WrappedComponent) {
+    class LocalizationWrapper extends React.Component {
+        componentDidUpdate (prevProps) {
```

Comment:
I'm still a little shaky on the react component lifecycle, but am i right in thinking that componentDidUpdate won't get called on the initial construct/mount/render? If so, onSetLanguage won't get called here when the page initially loads, right? Is that ok?

Reply:
I'll understand React better by looking at it running this code after it lands!